python/ray/tune/integration/docker.py

```diff
@@ -0,0 +1,187 @@
+import os
+import subprocess
+from typing import Any, Dict, List, Optional, Tuple
+
+import yaml
+from ray import services
+from ray.autoscaler._private.command_runner import DockerCommandRunner
+from ray.autoscaler.tags import NODE_KIND_HEAD, NODE_KIND_UNMANAGED, \
+    NODE_KIND_WORKER, TAG_RAY_NODE_KIND
+from ray.tune.syncer import NodeSyncer
+from ray.tune.sync_client import SyncClient
+
+
+class _WrappedProvider:
+    def __init__(self, config_path: str):
+        from ray.autoscaler._private.util import validate_config
+        from ray.autoscaler.node_provider import _get_node_provider
+
+        with open(config_path) as f:
+            new_config = yaml.safe_load(f.read())
+        validate_config(new_config)
+        self._config = new_config
```

Comment:
FYI @ericl for Tune to sync between docker containers, we leverage some of the autoscaler's private interfaces.

Reply:
Please advise if there's a better alternative

Reply:
Can you instead add validate_config to the public API (sdk.py)?